src/inmanta/agent/cache.py

```diff
@@ -307,6 +311,69 @@ def get_or_else(
             entering the cache.
 
         """
+
+        def _get_retention_policy(
```

Comment:
We now have two copies of this code in the code base, which will make it hard to keep the logic consistent across the two methods. We should get rid of that duplication.

Reply:
I removed it from the handler